packages/icons/src/constants.ts

```diff
@@ -19,7 +19,35 @@ export const ICON_V1_TEMPLATE = `<svg xmlns="http://www.w3.org/2000/svg">
   {uses}
 </svg>`;
 
+const ICON_V2_CONTAINER_STYLES = ICON_V2_SIZES.map((size, i) => {
+  const nextSize = ICON_V2_SIZES[i + 1];
+  const min = i !== 0 && `(min-width: ${size}px)`;
+  const max = i < ICON_V2_SIZES.length - 1 && `(max-width: ${nextSize - 1}px)`;
+
+  return `@container pi ${[min, max].filter(Boolean).join(' and ')} {
+    g {
+      --pis-${size}: block;
+    }
+  }`;
```

Comment:
Can you add a fallback for browsers that don't support container queries?

Reply:
Not necessary anymore, because I switched to use media queries. They are working great, and are supported since almost a decade. Please have a look at the tested browsers at the top of this PR.